common/graphics/texture.cc

```diff
@@ -26,7 +26,13 @@ void TextureRegistry::RegisterTexture(const std::shared_ptr<Texture>& texture) {
 void TextureRegistry::RegisterContextListener(
     uintptr_t id,
     std::weak_ptr<ContextListener> image) {
-  images_[id] = std::move(image);
+  size_t next_id = image_counter_.fetch_add(1);
```

Comment:
C++ operator overloading of `++` does this for you already. IMO that is more readable that `fetch_add`.

Reply:
Besides, does this need to be an atomic? I thought all texture registry operations were [raster thread only](https://github.com/flutter/engine/blob/6069c8f885a1bd829c26a66df8807165c7ad691c/common/graphics/texture.h#L77).

Reply:
I guess it doesn't need to be atomic. I was being overly cautious :)